src/python/dxpy/scripts/dx.py

```diff
@@ -3787,6 +3787,117 @@ def publish(args):
     except:
         err_exit()
 
+def archive(args):
+    dx_archive_errors = [InvalidState, ResourceNotFound, PermissionDenied]
+
+    # resolve paths
+    target_project = None    
+    target_folder = None
+    target_files = []
```

Comment:
We should think about the implications of two things if we already haven't (apologies in advanced if I missed a convo on this earlier): 

- What are practical limits to the length of `target_files` if any?
- What will be the system behavior if a user e.g. writes a shell script around `dx archive` on a metric-ton of files?  Should we design some 'in-product guidance' around that case? (@emiloslavsky  interest)

Reply:
> What are practical limits to the length of target_files if any? Should we design some 'in-product guidance' around that case? (@emiloslavsky interest)
The limit is defined in the API config, which is 1000 files now. For archival the default limit is hardcoded as 1000 and going over the limit would get us an apierror "InvalidInput". For unarchival

Reply:
* project-xxxx/archive (https://documentation.dnanexus.com/developer/api/data-containers/projects#api-method-project-xxxx-archive) does not  have a documented limit on the number of files.  If there's a limit, let's document it in project-xxxx/archive and in `dx archive` 
* Re "For unarchival, there is no default limit. And it would only unarchive the first N files and leave the rest untouched without raising an error" - not sure I'm following.  What is `N`?

Reply:
> * project-xxxx/archive (https://documentation.dnanexus.com/developer/api/data-containers/projects#api-method-project-xxxx-archive) does not have a documented limit on the number of files.  If there's a limit, let's document it in project-xxxx/archive and in `dx archive`
The limit is coded in API server config or hardcoded in the code. I will put it in the documentation and in the cli help.
> * Re "For unarchival, there is no default limit. And it would only unarchive the first N files and leave the rest untouched without raising an error" - not sure I'm following.  What is `N`?
The 'N' is the number above (the one in config; if none then 1000)